python/wheelbuilder/Dockerfile

```diff
@@ -0,0 +1,98 @@
+ENV PERL_VERSION 5.28.0
+RUN curl -O https://www.cpan.org/src/5.0/perl-${PERL_VERSION}.tar.gz
+RUN tar xzf perl-${PERL_VERSION}.tar.gz
+RUN cd perl-${PERL_VERSION} && ./Configure -de
+RUN cd perl-${PERL_VERSION} && make
+RUN cd perl-${PERL_VERSION} && make install
+
+# Install OpenSSL
+ENV OPENSSL_VERSION 1.1.1a
+RUN curl -O https://www.openssl.org/source/openssl-${OPENSSL_VERSION}.tar.gz
+RUN tar xzf openssl-${OPENSSL_VERSION}.tar.gz
+RUN cd openssl-${OPENSSL_VERSION} && \
+    ./config shared no-ssl2 no-ssl3 -fPIC --prefix=/usr/local
+RUN cd openssl-${OPENSSL_VERSION} && \
+    sed -i "s/^SHLIB_MAJOR=.*/SHLIB_MAJOR=200/" Makefile && \
+    sed -i "s/^SHLIB_MINOR=.*/SHLIB_MINOR=0.0/" Makefile && \
+    sed -i "s/^SHLIB_VERSION_NUMBER=.*/SHLIB_VERSION_NUMBER=200.0.0/" Makefile
+RUN cd openssl-${OPENSSL_VERSION} && make depend
+RUN cd openssl-${OPENSSL_VERSION} && make
+RUN cd openssl-${OPENSSL_VERSION} && make install_sw install_ssldirs
+
+RUN ldconfig -v
+
+# Install cyrus-sasl
+ENV SASL_VERSION 2.1.27
+RUN curl -O ftp://ftp.cyrusimap.org/cyrus-sasl/cyrus-sasl-${SASL_VERSION}.tar.gz
+RUN tar zxf cyrus-sasl-${SASL_VERSION}.tar.gz
+RUN cd cyrus-sasl-${SASL_VERSION} && \
+    ./configure --disable-silent-rules \
+                --disable-cmulocal \
+                --disable-sample \
+                --disable-obsolete_cram_attr \
+                --disable-obsolete_digest_attr \
+                --disable-staticdlopen \
+                --disable-java \
+                --disable-alwaystrue \
+                --enable-checkapop \
+                --enable-cram \
+                --enable-digest \
+                --enable-scram \
+                --disable-otp \
+                --disable-srp \
+                --disable-srp-setpass \
+                --disable-krb4 \
+                --disable-gssapi \
+                --disable-gss_mutexes \
+                --disable-sia \
+                --disable-auth-sasldb \
+                --disable-httpform \
+                --enable-plain \
+                --enable-anon \
+                --enable-login \
+                --disable-ntlm \
+                --disable-passdss \
+                --disable-sql \
+                --disable-ldapdb \
+                --disable-macos-framework
+RUN cd cyrus-sasl-${SASL_VERSION} && make
+RUN cd cyrus-sasl-${SASL_VERSION} && make install
+RUN ldconfig -v
+
+# Install libetpan
+ENV ETPAN_VERSION 1.9.1
+RUN curl -L -o libetpan-${ETPAN_VERSION}.tar.gz \
+    https://github.com/dinhviethoa/libetpan/archive/${ETPAN_VERSION}.tar.gz
+RUN tar xzf libetpan-${ETPAN_VERSION}.tar.gz
+RUN cd libetpan-${ETPAN_VERSION} && ./autogen.sh && \
+    ./configure --enable-ipv6 \
+                --disable-iconv --disable-db \
+                --with-openssl --with-sasl --with-zlib \
+                --without-curl --without-expat
+RUN cd libetpan-${ETPAN_VERSION} && make
+RUN cd libetpan-${ETPAN_VERSION} && make install
+RUN ldconfig -v
```

Comment:
Also an example here https://github.com/deltachat/docker-images/blob/master/debian-stretch-node-11/Dockerfile

Reply:
Maybe you don't want to put _everything_ into _one_ `RUN` but it could make sense to use one `RUN` per 'artifact', e.g. one for `libetpan`, one for `sasl` etc.

Reply:
Yes, I'm aware of how RUN behaves like this.  I actually explicitly want the intermediate steps to be cached, i.e. have the downloads not repeated for build options changes etc.  There's a balance somewhere I guess and I decided this was appropriate here.  Also I'm of the opinion that one should try to separate build images from prod images, and this is a build image so the size and intermediate artefacts left around are not a concern (often even desired as I said above).

Reply:
> Also I'm of the opinion that one should try to separate build images from prod images, and this is a build image so the size and intermediate artefacts left around are not a concern (often even desired as I said above).

That's a good point and I agree.